src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -42,21 +48,29 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
         // this.pageSize = queryParams['size'] ? Number(queryParams['size']) : 10;
         this.requestComplexResults();
         document.body.scrollTop = 0;
+        console.log("test" + this._allComponentsInComplexSearch);
       });
+
   }
 
   ngAfterViewInit(): void {
   }
 
   private requestComplexResults() {
-    this.complexPortalService.findComplex(this.query, this.spicesFilter, this.bioRoleFilter,
+    // Calling new HACKY method to get also the components of each complex
+    this.complexPortalService.findComplexWithComponents(this.query, this.spicesFilter, this.bioRoleFilter,
       this.interactorTypeFilter, this.currentPageIndex, this.pageSize).subscribe(complexSearch => {
       this.complexSearch = complexSearch;
```

Comment:
Here we need to call `this.allComponentsInComplexSearch = new Set();` to empty the set before setting all the components in it.